umrr_ros2_driver/src/smartmicro_radar_node.cpp

```diff
@@ -166,6 +171,126 @@ SmartmicroRadarNode::SmartmicroRadarNode(const rclcpp::NodeOptions & node_option
     m_publishers[i] = create_publisher<sensor_msgs::msg::PointCloud2>(
       "umrr/targets_" + std::to_string(i), sensor.history_size);
   }
+
+  // create a ros2 service to change the radar parameters
+  command_srv_ = create_service<umrr_ros2_msgs::srv::SetParam>(
+    "smartmicro_radar_node/set_radar_parameter",
+    std::bind(
+      &SmartmicroRadarNode::change_radar_mode, this, std::placeholders::_1,
+      std::placeholders::_2));
+  RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Radar parameter services are ready.");
+
+  // create a ros2 service to change the IP address
+  ip_addr_srv_ = create_service<umrr_ros2_msgs::srv::SetIp>(
+    "smartmicro_radar_node/configure_ip_address",
+    std::bind(
+      &SmartmicroRadarNode::change_ip_address, this, std::placeholders::_1,
+      std::placeholders::_2));
+  RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "Radar is ready to be configured.");
+}
+
+void SmartmicroRadarNode::change_radar_mode(
+  const std::shared_ptr<umrr_ros2_msgs::srv::SetParam::Request> request,
+  std::shared_ptr<umrr_ros2_msgs::srv::SetParam::Response> result)
+{
+  std::shared_ptr<InstructionServiceIface> inst{m_services->GetInstructionService()};
+  std::this_thread::sleep_for(std::chrono::seconds(2));
```

Comment:
This is bad practice. Since the callbacks on the node are processed in a queue sequentially (unless using a MultiThreadedExecutor), this will cause your node to be unresponsive during these 2 seconds. I suggest finding another way to handle this like creating a one-shot timer which triggers two seconds from the time that this call comes in. e.g. `my_timer = this->create_wall_timer(std::chrono::seconds(2), std::bind(my_timer_callback))`. Then, inside the callback, call `my_timer->cancel()` which will make it run only once.